trees/trees_test.go

```diff
@@ -0,0 +1,369 @@
+		opts                                 GetOpts
+		ctxTree, storageTree, wantTree       *trillian.Tree
+		beginErr, getErr, commitErr, wantErr bool
+	}{
+		{
+			desc:        "logTree",
+			treeID:      logTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG},
+			storageTree: &logTree,
+			wantTree:    &logTree,
+		},
+		{
+			desc:        "mapTree",
+			treeID:      mapTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_MAP},
+			storageTree: &mapTree,
+			wantTree:    &mapTree,
+		},
+		{
+			desc:        "wrongType1",
+			treeID:      logTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_MAP},
+			storageTree: &logTree,
+			wantErr:     true,
+		},
+		{
+			desc:        "wrongType2",
+			treeID:      mapTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG},
+			storageTree: &mapTree,
+			wantErr:     true,
+		},
+		{
+			desc:        "frozenTree",
+			treeID:      frozenTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG, Readonly: true},
+			storageTree: &frozenTree,
+			wantTree:    &frozenTree,
+		},
+		{
+			desc:        "frozenTreeNotReadonly",
+			treeID:      frozenTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG},
+			storageTree: &frozenTree,
+			wantErr:     true,
+		},
+		{
+			desc:        "softDeleted",
+			treeID:      softDeletedTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG},
+			storageTree: &softDeletedTree,
+			wantErr:     true,
+		},
+		{
+			desc:        "hardDeleted",
+			treeID:      hardDeletedTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG},
+			storageTree: &hardDeletedTree,
+			wantErr:     true,
+		},
+		{
+			desc:     "treeInCtx",
+			treeID:   logTree.TreeId,
+			opts:     GetOpts{TreeType: trillian.TreeType_LOG},
+			ctxTree:  &logTree,
+			wantTree: &logTree,
+		},
+		{
+			desc:        "wrongTreeInCtx",
+			treeID:      logTree.TreeId,
+			opts:        GetOpts{TreeType: trillian.TreeType_LOG},
+			ctxTree:     &mapTree,
+			storageTree: &logTree,
+			wantTree:    &logTree,
+		},
+		{
+			desc:     "beginErr",
+			treeID:   logTree.TreeId,
+			opts:     GetOpts{TreeType: trillian.TreeType_LOG},
+			beginErr: true,
+			wantErr:  true,
+		},
+		{
+			desc:    "getErr",
+			treeID:  logTree.TreeId,
+			opts:    GetOpts{TreeType: trillian.TreeType_LOG},
+			getErr:  true,
+			wantErr: true,
+		},
+		{
+			desc:      "commitErr",
+			treeID:    logTree.TreeId,
+			opts:      GetOpts{TreeType: trillian.TreeType_LOG},
+			commitErr: true,
+			wantErr:   true,
+		},
+	}
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	for _, test := range tests {
+		ctx := NewContext(context.Background(), test.ctxTree)
+
+		var beginErr, getErr, commitErr error
+		if test.beginErr {
+			beginErr = errors.New("begin error")
+		}
+		if test.getErr {
+			getErr = errors.New("get error")
+		}
+		if test.commitErr {
+			commitErr = errors.New("commit error")
+		}
```

Comment:
why not make the fields in `tests` of type `error` and skip the fixed mapping from `bool` to `error`?